pageserver/src/layered_repository/disk_btree.rs

```diff
@@ -106,12 +123,12 @@ impl<'a, const L: usize> OnDiskNode<'a, L> {
     ///
     /// Interpret a PAGE_SZ page as a node.
     ///
-    fn deparse(buf: &[u8]) -> OnDiskNode<L> {
+    fn deparse(buf: &[u8]) -> Result<OnDiskNode<L>> {
```

Comment:
Wdyt about leaving `Result` for `std::result::Result` that's already implicitly imported by the std prelude and qualify other results?
We have code places where `Result` is `anyhow::Result`, now we have other, per-module results, with all the `?` it might be hard to use these on a higher level?

I would personally just use `-> anyhow::Result<T>` for anyhow and `-> Result<T, E>` for thiserror and the rest, it's not much code to write, but makes things a bit more readable IMO.

Reply:
How would you call a result alias defined [near the error type](https://github.com/neondatabase/neon/pull/1612/files#diff-259bfb5a4e25aaa6c1655af0dc3777e277ef986d0cd4e396bb5b642dc8d8a20aR104) then? I think it has a bit higher priority than the std result. Another option is to put it to a `mod something` e g `mod error` and leave the std result as is. This is the same way how it is structured in std::io. There is a type alias for the result here https://doc.rust-lang.org/src/std/io/error.rs.html#55 and then functions return just `Result<T`> e g [fs::read](https://doc.rust-lang.org/std/fs/fn.read.html#). I'm ok to change that if you insist. That saves one `Go to definition` skipping through result alias if you want to get to the error type definition

Reply:
Definitely not insisting, just curious if the new approach is 

I find the `-> Result<Val>` quite confusing since it's hard to know which result that is on some line 472, especially when reviewing the code without IDE.
I was thinking, since we define `E` type with `thiserror` anyway, to use basic `std::result::Result<Val, CustomE>` everywhere in the new places to make it clearer to see what happens in the module and around it.
The old code would live as is, with maybe `anyhow::Result` instead of `Result` used when possible.

But no strong opinions, go ahead and merge it as is to continue the experiment unchanged.